src/cowboy_telemetry_h.erl

```diff
@@ -0,0 +1,57 @@
+-module(cowboy_telemetry_h).
+-behavior(cowboy_stream).
+
+-export([init/3]).
+-export([data/4]).
+-export([info/3]).
+-export([terminate/3]).
+-export([early_error/5]).
+
+init(StreamID, Req, Opts) ->
+  SystemTime = erlang:system_time(),
+  StartTime = erlang:monotonic_time(),
+  telemetry:execute(
+    [cowboy, request, start],
+    #{system_time => SystemTime},
+    #{stream_id => StreamID, req => Req}
+  ),
+  {Commands, Next} = cowboy_stream:init(StreamID, Req, Opts),
+  {Commands, [Next, StartTime]}.
```

Comment:
Nitpick: it should be `[Next | StartTime]` (for roflscale reasons). A tuple is also fine.